### PR TITLE
[Test] Add unit tests for managers/scheduler_components/logprob_result_processor

### PR DESCRIPTION
**Pull request by w2315516642**
## Motivation

Add focused CPU-only unit test for
`managers/scheduler_components/logprob_result_processor.py`. Part of #20865.

## Modifications

Add `unit/managers/test_logprob_result_processor.py` which contains 5 tests for source file `logprob_result_processor.py`. No server launch, no model weight loading.

These tests cover the main result-processing paths:

- Regular input logprobs are aligned with prompt tokens, with the sampling token removed and invalid vocab ids clipped.
- `top_logprobs` and `token_ids_logprob` are aligned with input tokens and their temporary buffers are cleared.
- MIS delimiter logprobs are kept unchanged instead of being shifted like regular input logprobs.
- Output logprobs are appended even when no input logprobs are returned.
- Input-logprob counts use all positions for regular requests and only delimiter positions for MIS requests.

## Accuracy Tests

N/A. This is a test-only change and does not modify model outputs or inference behavior.

## Speed Tests and Profiling

N/A. This is a CPU-only unit-test addition and does not modify inference performance.

## Validation

The targeted test file passes locally. Coverage for the target source file reaches 100% statement coverage.
```
Name                                                                          Stmts   Miss  Cover   Missing
-----------------------------------------------------------------------------------------------------------
python/sglang/srt/managers/scheduler_components/logprob_result_processor.py     139      0   100%
-----------------------------------------------------------------------------------------------------------
TOTAL                                                                           139      0   100%
```

```
# command
pytest test/registered/unit/managers/test_logprob_result_processor.py -v
# output
=========================================================== test session starts ============================================================
platform linux -- Python 3.12.13, pytest-9.1.1, pluggy-1.6.0 -- /root/miniconda3/envs/sglang/bin/python
cachedir: .pytest_cache
rootdir: /root/autodl-tmp/sglang/test
configfile: pytest.ini
plugins: cov-7.1.0, anyio-4.14.2
collected 5 items                                                                                                                          

test/registered/unit/managers/test_logprob_result_processor.py::TestSchedulerLogprobResultProcessor::test_add_logprob_return_values_initializes_empty_input_containers PASSED [ 20%]
test/registered/unit/managers/test_logprob_result_processor.py::TestSchedulerLogprobResultProcessor::test_calculate_num_input_logprobs_regular_and_mis PASSED [ 40%]
test/registered/unit/managers/test_logprob_result_processor.py::TestSchedulerLogprobResultProcessor::test_multi_item_scoring_keeps_delimiter_logprobs_without_leading_none PASSED [ 60%]
test/registered/unit/managers/test_logprob_result_processor.py::TestSchedulerLogprobResultProcessor::test_regular_input_logprobs_align_and_clip_vocab_boundary PASSED [ 80%]
test/registered/unit/managers/test_logprob_result_processor.py::TestSchedulerLogprobResultProcessor::test_top_and_token_ids_logprobs_drop_sampling_token_and_clear_temp PASSED [100%]

============================================================= warnings summary =============================================================
../../miniconda3/envs/sglang/lib/python3.12/site-packages/_pytest/config/__init__.py:1464
  /root/miniconda3/envs/sglang/lib/python3.12/site-packages/_pytest/config/__init__.py:1464: PytestConfigWarning: Unknown config option: asyncio_mode
  
    self._warn_or_fail_if_strict(f"Unknown config option: {key}\n")

../../miniconda3/envs/sglang/lib/python3.12/site-packages/torch/jit/_script.py:365: 14 warnings
  /root/miniconda3/envs/sglang/lib/python3.12/site-packages/torch/jit/_script.py:365: DeprecationWarning: `torch.jit.script_method` is deprecated. Please switch to `torch.compile` or `torch.export`.
    warnings.warn(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
====================================================== 5 passed, 15 warnings in 9.53s ======================================================
```

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

---
### CI States

Latest PR Test (Base): :x: [Run #30470663110](https://github.com/sgl-project/sglang/actions/runs/30470663110)
Latest PR Test (Extra): :x: [Run #30470664125](https://github.com/sgl-project/sglang/actions/runs/30470664125)

### CODE REVIEW

**Comment by hnyls2002** [comment]
Closing this along with the other open unit-test PRs referenced from #20865.

In the AI era, adding unit tests for coverage's sake no longer provides enough value to justify the cost — each PR in this batch consumes reviewer bandwidth and CI time out of proportion to the coverage it adds. The tracking issue itself is already closed, and we are no longer accepting coverage-driven UT PRs from it.

Thanks for the effort — test contributions tied to a concrete bug fix or behavior change are still welcome.
